packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -164,11 +170,68 @@ export class LedgerKeyAgent extends KeyAgentBase {
     });
   }
 
+  async signTransaction(initializedTxData: InitializeTxResult): Promise<Cardano.Signatures> {
+    try {
+      const deviceConnection = await LedgerKeyAgent.checkDeviceConnection(this.serializableData.communicationType, this.deviceConnection);
+      const dataForSigning = this.#prepareDataForSigning(initializedTxData);
+      const response = await deviceConnection.signTransaction(dataForSigning);
+      // TODO - derive witnes(es) path(s) and return Map<Ed25519PublicKey, Ed25519Signature>
+      return response.witnesses;
+    } catch (error: any) {
+      // eslint-disable-next-line unicorn/numeric-separators-style
+      if (error.code === 28169) {
+        throw new AuthenticationError('Transaction signing aborted', error);
+      }
+      throw new TransportError('Transport failed', error);
+    }
+    return response;
+  }
+
   async signBlob(): Promise<SignBlobResult> {
     throw new NotImplementedError('signBlob');
   }
 
   async exportRootPrivateKey(): Promise<Cardano.Bip32PrivateKey> {
     throw new NotImplementedError('Operation not supported!');
   }
+
+  async #prepareDataForSigning({ body, hash, inputSelection }: InitializeTxResult) {
+    // TODO - Isolate change outputs
+    // TODO - Prepare token bundle
+    const inputs = body.inputs.map((input) => ({
+      txHashHex: input.txId,
+      outputIndex: input.txIndex,
+      // TODO - Derivation Path - get from `ownSignatureKeyPaths`
+      path: utils.str_to_path("1852'/1815'/0'/0/0"),
+    }));
+
+    const outputs = body.outputs.map((output) => ({
+      amount: output.value.coins,
+      tokenBundle: output.value.assets || [], // TODO - Check & Calculate
+      destination: {
+        // TODO - remove address type declaration comments
+        // DEVICE_OWNED - for change address
+        // THIRD_PARTY - for regular output
+        type: TxOutputDestinationType.THIRD_PARTY,
+        params: {
+          addressHex: utils.buf_to_hex(utils.bech32_decodeAddress(output.address))
+        },
+      },
+    }));
+
+    return {
+      tx: {
+        network: {
+          networkId: this.serializableData.networkId,
+          protocolMagic: 42, // TODO - replace with real magic - Faked to Testnet 
```

Comment:
This can be a prop when creating key agent

Reply:
cool, I was thinking about constant because protocolMagic depends on `networdId` and automatically return it regarding passed network. What you think about that @rhyslbw @mkazlauskas?

Reply:
That sounds ok to me but don't really know what it the magic is

Reply:
It could be e.g.
```
// Cardano Byron Testnet network magic
  [TESTNET]: [1097911063, 0],
  // Cardano Alonzo Purple network magic
  [ALONZO_PURPLE]: [8, 0],
  // Cardano Shelley QA network magic
  [SHELLEY_QA]: [3, 0],
```

Reply:
Yeah network ID is a network _type_ identifier, telling us if it's the mainnet or **a** testnet. Network magic is the network identifier. This should really come from a provider, so we can use a remote service to provision the key agent during a handshake if that's an option.